lib/state/appActions.js

```diff
@@ -236,16 +236,39 @@ appActions.login = function(credentials, options, cb) {
   });
 };
 
+//make sure we have the correct default targetId
+appActions._getDefaultTargetId = function(loggedInUser){
+  //default to the logged in users id
+  var userid = loggedInUser.userid;
+
+  //even though we have multiple groups there may still only be one upload account
+  if (_.isEmpty(loggedInUser.uploadGroups) === false || loggedInUser.uploadGroups.length > 1) {
+
+    var possibilities = _.filter(loggedInUser.uploadGroups, function(group) {
+      //if the account isn't marked with patient details we don't upload to it
+      return _.isEmpty(group.profile) === false || _.isEmpty(group.profile.patient) === false;
```

Comment:
I don't think this logic is right... As far as I can tell, `group.profile` is never empty, so you're always returning `true` in this filter and not filtering anything out.